enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -561,7 +561,7 @@ private MasterClient newMasterClient( URI masterUri, URI me, StoreId storeId, Li
     {
         MasterClient masterClient = masterClientResolver.instantiate( masterUri.getHost(), masterUri.getPort(),
                 me.getHost(), monitors, storeId, life );
-        if ( masterClient.getProtocolVersion().compareTo( MasterClient210.PROTOCOL_VERSION ) < 0 )
+        if ( masterClient.getProtocolVersion().compareTo( MasterClient214.PROTOCOL_VERSION ) < 0 )
         {
             idGeneratorFactory.enableCompatibilityMode();
```

Comment:
Since we now have removed `MasterClient210`, then `HaIdGeneratorFactory#enableCompatibilityMode()` should also go since it was a work-around for cross-2.0->2.1 cluster migrations. `MasterClient214` has a misleading name, it's when it was created, but was actually used up until now. This condition and the compatibility mode should be removed now.